enderio-conduits/src/main/java/crazypants/enderio/conduits/conduit/liquid/EnderLiquidConduitNetwork.java

```diff
@@ -46,9 +46,13 @@ public void connectionChanged(@Nonnull EnderLiquidConduit con, @Nonnull EnumFaci
       sort = true;
     }
 
-    tanks.remove(tank); // remove old tank, NB: =/hash is only calced on location and dir
+    int oldIndex = tanks.indexOf(oldTank);
+    if (oldIndex == -1) {
+      tanks.add(tank);
```

Comment:
wouldn't it need to sort the list in this case, too?

(I wonder if using a self-sorting list here would not be the better solution.)

Reply:
I get the feeling this method is trying to be too clever. Let's just sort the list unconditionally on each change.

Reply:
Exactly what I was thinking.